src/server/game/Maps/Map.cpp

```diff
@@ -3036,7 +3036,10 @@ void InstanceMap::RemovePlayerFromMap(Player* player, bool remove)
     //if (!m_unloadTimer && m_mapRefMgr.getSize() == 1)
     //    m_unloadTimer = m_unloadWhenEmpty ? MIN_UNLOAD_DELAY : std::max(sWorld->getIntConfig(CONFIG_INSTANCE_UNLOAD_DELAY), (uint32)MIN_UNLOAD_DELAY);
     Map::RemovePlayerFromMap(player, remove);
-    player->SetPendingBind(0, 0);
+
+    // If remove == true - player already deleted.
+    if (!remove)
+        player->SetPendingBind(0, 0);
```

Comment:
We still need to clear pending binds, even if the player is removed.

We don't clear it in Map::RemovePlayerFromMap

Reply:
It doesn't make sense for a deleted player. Why should we do it? When I say `deleted`, I mean literally deleted object, like `delete player;`.

Reply:
you are right, I was looking at it wrong